annif/corpus/document.py

```diff
@@ -90,20 +90,21 @@ def documents(self):
         yield from self._documents
 
 
-class TruncatingDocumentCorpus(DocumentCorpus):
-    """A document corpus that wraps another document corpus but truncates the
-    documents to a given length"""
+class TransformingDocumentCorpus(DocumentCorpus):
+    """A document corpus that wraps another document corpus but transforms the
+    documents using a given transform function"""
 
-    def __init__(self, corpus, limit):
+    def __init__(self, corpus, transform_text):
```

Comment:
How about calling the parameter `transform_function` or `transform_fn` or something along those lines? It's not obvious that `transform_text` should be a function.

I suggest adding a docstring to the constructor as well, clarifying the meaning of the parameters.